src/data/automation.ts

```diff
@@ -109,10 +109,17 @@ export interface TemplateTrigger {
   value_template: string;
 }
 
+export interface ContextConstraint {
+  context_id: string;
+  parent_id?: string;
+  user_id?: string | string[];
+}
+
 export interface EventTrigger {
   platform: "event";
   event_type: string;
   event_data: any;
+  context: ContextConstraint;
```

Comment:
```suggestion
  event_data?: any;
  context?: ContextConstraint;
```
And you should guard for context being undefined.